src/wrappers/gdalGetRasterStatistics.js

```diff
@@ -0,0 +1,63 @@
+/* global Module */
+export default function (GDALGetRasterStatistics, errorHandling) {
+    return function (datasetPtr, bandNum) {
+        const bandPtr = Module.ccall(
+            'GDALGetRasterBand',
+            'number',
+            ['number', 'number'],
+            [datasetPtr, bandNum]
+        );
+        // We need to allocate pointers to store statistics into which will get passed into
+        // GDALGetRasterStatistics(). They're all doubles, so allocate 8 bytes each.
+        const minPtr = Module._malloc(8);
+        const maxPtr = Module._malloc(8);
+        const meanPtr = Module._malloc(8);
+        const sdPtr = Module._malloc(8);
+        const returnErr = GDALGetRasterStatistics(
+            bandPtr,
+            0, // Approximate statistics flag -- set to false
+            1, // Force flag -- will always return statistics even if image must be rescanned
+            minPtr,
+            maxPtr,
+            meanPtr,
+            sdPtr
+        );
+
+        const errorType = errorHandling.CPLGetLastErrorType();
+
+        // This is our cleanup, but the ordering changes depending on whether this is a failure case
+        // or not, so it's broken out into a function so that it's always the same when we call it.
+        function cleanup() {
+            Module._free(minPtr);
+            Module._free(maxPtr);
+            Module._free(meanPtr);
+            Module._free(sdPtr);
+        }
+        // Check for errors; throw if error is detected
+        // GDALGetRasterStatistics returns CE_Failure if an error occurs.
+        if (
+            errorType === errorHandling.CPLErr.CEFailure ||
+            errorType === errorHandling.CPLErr.CEFatal ||
+            returnErr === errorHandling.CPLErr.CEFailure
+        ) {
+            let message = errorHandling.CPLGetLastErrorMsg();
+
+            cleanup();
+
+            throw new Error('Error in GDALGetRasterStatistics: ' + message);
+        } else {
+            // At this point the values at each pointer should have been written with statistics so
+            // we can read them out and send them back.
+
+            const result = {
+                minimum: Module.getValue(minPtr, 'double'),
+                maximum: Module.getValue(maxPtr, 'double'),
+                median: Module.getValue(meanPtr, 'double'),
+                stdDev: Module.getValue(sdPtr, 'double'),
+            };
+
+            cleanup();
+            return result;
+        }
```

Comment:
`try`/`finally` might work here:
```suggestion
        try {
            // Check for errors; throw if error is detected
            // GDALGetRasterStatistics returns CE_Failure if an error occurs.
            if (
                errorType === errorHandling.CPLErr.CEFailure ||
                errorType === errorHandling.CPLErr.CEFatal ||
                returnErr === errorHandling.CPLErr.CEFailure
            ) {
                let message = errorHandling.CPLGetLastErrorMsg();

                throw new Error('Error in GDALGetRasterStatistics: ' + message);
            } else {
                // At this point the values at each pointer should have been written with statistics so
                // we can read them out and send them back.

               return {
                    minimum: Module.getValue(minPtr, 'double'),
                    maximum: Module.getValue(maxPtr, 'double'),
                    median: Module.getValue(meanPtr, 'double'),
                    stdDev: Module.getValue(sdPtr, 'double'),
                };
            }
        } finally {
            Module._free(minPtr);
            Module._free(maxPtr);
            Module._free(meanPtr);
            Module._free(sdPtr);
        }
```

Reply:
Ooh, great idea! This is exactly how I would have done it in Python, I have no idea why that never occurred to me here. 🤦 I'm going to give this a shot for one of the functions here, and assuming it works as expected I'm going to open an issue to move the whole codebase over, because this pattern gets repeated a lot.

Reply:
This worked great and simplified things a lot! I opened #113 to move the rest of the codebase over.